pkg/motor/query.go

```diff
@@ -0,0 +1,121 @@
+package motor
+
+import (
+	"fmt"
+
+	"github.com/sonr-io/sonr/thirdparty/types/common"
+	ct "github.com/sonr-io/sonr/thirdparty/types/common"
+	mt "github.com/sonr-io/sonr/thirdparty/types/motor"
+	bt "github.com/sonr-io/sonr/x/bucket/types"
+)
+
+func (mtr *motorNodeImpl) Query(req mt.QueryRequest) (mt.QueryResponse, error) {
+	results := make([]*mt.QueryResultItem, 0)
+	switch req.GetModule() {
+	case ct.BlockchainModule_SCHEMA:
+		i, err := mtr.handleSchemaQuery(req.GetQuery())
+		if err != nil {
+			return mt.QueryResponse{}, err
+		}
+		results = append(results, i)
+	case ct.BlockchainModule_REGISTRY:
+		i, err := mtr.handleRegistryQuery(req.GetQuery())
+		if err != nil {
+			return mt.QueryResponse{}, err
+		}
+		results = append(results, i)
+	case ct.BlockchainModule_BUCKET:
+		if req.GetKind() == common.EntityKind_ADDRESS {
+
+			i, err := mtr.handleBucketGroupQuery(req.GetQuery())
+			if err != nil {
+				return mt.QueryResponse{}, err
+			}
+			results = append(results, i)
+
+		} else {
+			i, err := mtr.handleBucketQuery(req.GetQuery())
+			if err != nil {
+				return mt.QueryResponse{}, err
+			}
+			results = append(results, i)
+
+		}
+	}
+
+	return mt.QueryResponse{
+		Results: results,
+		Query:   req.GetQuery(),
+		Module:  req.GetModule(),
+	}, nil
+}
+
+func (mtr *motorNodeImpl) handleBucketQuery(did string) (*mt.QueryResultItem, error) {
+	// use the item within the cache from GetWhereIs
+	if wi := mtr.Resources.whereIsStore[did]; wi != nil {
```

Comment:
Use the getter and setter methods in `Resources` instead of manually modifying

Reply:
Fixed